### DISCUSSION TITLE
There are some issues with RAG full-text search

### DISCUSSION THREAD

**Discussion by lai-serena** [langgenius/dify#4890] [General]
I used xinreference to integrate the BGE or BCE embedding model and added some knowledge to the knowledge base. Both vector search and hybrid search yielded results, but full-text search yielded no search results. Why is this?
This is data in the knowledge base and there is a "黑名单"
![1717409705933](https://github.com/langgenius/dify/assets/44218061/25cfdeb2-01cb-4694-a5ca-83153022c9db)
But when using full-text search, there were no results
![1717409741234](https://github.com/langgenius/dify/assets/44218061/5d8010ee-690f-408b-a106-7a14b0aa3cdf)

**Comment by patryk20120** [comment]
It depends how you search for the text. All 3 options work a little bit differently from each other and the result depends on input search.

**Reply by lai-serena** [reply]
Thank you for your answer. But I am really confused about what kind of input will result in full-text search. Does this mean that the input should have more details and text?

**Reply by patryk20120** [reply]
Well, this might help you out https://docs.dify.ai/features/retrieval-augment/hybrid-search#full-text-search

Can't read your language so I don't even know how the chunk looks and what you are looking for.

**Comment by hanqingwu** [comment]
yes,  when I use Chinese character , most full text search return none records .
